h/views/admin_groups.py

```diff
@@ -150,6 +168,7 @@ def on_success(appstruct):
             group.description = appstruct['description']
             group.name = appstruct['name']
             group.scopes = [GroupScope(origin=o) for o in appstruct['origins']]
+            group.organization = _find_organization(self.request.db, appstruct['organization'])
```

Comment:
That's not what I recall from the discussion around the [ADR when we introduced the service layer](https://github.com/hypothesis/h/blob/master/docs/arch/adr-002.rst#decision). My recollection was that we would use judgement to avoid putting "non trivial" model logic/queries in views, or when there is a clear advantage to testability or reusability.

Reply:
I think we should tend aggressively to moving as much as possible (sqlalchemy queries, and everything else possible) out of the views, because otherwise views tend to easily get long and complicated, and to have a lot of different responsibilities (receiving and validating data from the request, carrying out business logic, querying the DB, preparing the response, ...) and they also become very difficult to test. A lot of long and difficult to read tests in very long test modules. Witness the admin pages views, but _especially_ the activity pages views and their tests.

I think we should look at views as a thin-as-possible integration layer between the web framework and the rest of our code. Receive the request from the web framework, call other parts of our code (schemas, services, presenters, ...) to do the work, send the response back to the web framework. The views are the things that plug the different parts of our architecture together (taking the return result of schema validation and passing that into a service, etc) and plug our code in to Pyramid's request and response. But they don't do anything else - just dumb integration.

For sqlalchemy queries in particular: sqlalchemy queries are tightly coupled to the structure and characteristics (constraints, performance, ...) of our model classes and database tables and columns. Something like `db.query(models.Group).filter_by(pubid="__world__").one()` contains knowledge such as: `models.Group` exists, groups have a `pubid`, `__world__` is the special pubid of the default group, there's only one default group, retrieving a group by pubid is efficient, etc.

So we need to avoid peppering these queries throughout our entire codebase, otherwise we couple the whole codebase to the DB structure, and making any changes to the DB structure becomes difficult.

So keeping sqlalchemy queries within models and services.